pandas/core/construction.py

```diff
@@ -560,8 +560,11 @@ def sanitize_array(
             raise TypeError(f"'{type(data).__name__}' type is unordered")
 
         # materialize e.g. generators, convert e.g. tuples, abc.ValueView
-        # TODO: non-standard array-likes we can convert to ndarray more efficiently?
-        data = list(data)
+        if hasattr(data, "__array__"):
```

Comment:
can you run this thru L518? where it makes more sense

Reply:
will take some rearrangement (move EA check about ndarray check) but i think doable